File: backend/src/IssueTracker.Core/Entities/Issue.cs
using IssueTracker.Core.Enums;

namespace IssueTracker.Core.Entities;

/// <summary>
/// Represents an issue in the tracking system
/// </summary>
public class Issue
{
    /// <summary>
    /// Unique identifier for the issue
    /// </summary>
    public int Id { get; set; }

    /// <summary>
    /// Brief title describing the issue
    /// </summary>
    public string Title { get; set; } = string.Empty;

    /// <summary>
    /// Detailed description of the issue
    /// </summary>
    public string Description { get; set; } = string.Empty;

    /// <summary>
    /// Current status of the issue
    /// </summary>
    public IssueStatus Status { get; set; } = IssueStatus.Open;

    /// <summary>
    /// Date and time when the issue was created (UTC)
    /// </summary>
    public DateTime CreatedAt { get; set; }

    /// <summary>
    /// Date and time when the issue was resolved (UTC), null if not resolved
    /// </summary>
    public DateTime? ResolvedAt { get; set; }
}

File: frontend/src/types/domain/issue.ts
/**
 * Issue status enumeration
 */
export enum IssueStatus {
  Open = 1,
  InProgress = 2,
  Resolved = 3,
}

/**
 * Issue domain model
 * Represents an issue in the system
 */
export interface Issue {
  id: number;
  title: string;
  description: string;
  status: IssueStatus;
  createdAt: string;
  resolvedAt: string | null;
}
